GameOfLife.java

```diff
@@ -58,21 +67,56 @@ private static void play(String fileName) {
 	// and the rightmost columns. Thus the actual board is surrounded by a "frame" of zeros. You can think
 	// of this frame as representing the infinite number of dead cells that exist in every direction.
 	// This function assumes that the input file contains valid data, and does no input testing.
-	private static int[][] read(String fileName) {
+	public static int[][] read(String fileName) {
 		In in = new In(fileName); // Constructs an In object for reading the input file
 		int rows = Integer.parseInt(in.readLine());
 		int cols = Integer.parseInt(in.readLine());
 		int[][] board = new int[rows + 2][cols + 2];
+
+		String ones="";
+		while (in.isEmpty()==false && ones=="") {
+			ones = in.readLine();
+		}
+
+		int middleRow = rows/2;
+		if(rows%2==0){
+			middleRow = rows/2-1;
+		}
+		boolean nextLine= true;
+		for (int i = middleRow; i<rows; i++) {
+			for (int j = 0;j < cols && nextLine==true; j++) {
+				if(j<ones.length()){
+					if(ones.charAt(j)=='x'){
+						board[i][j]=1;
+					}
+				}	
+			}
+			if (in.isEmpty()==false) {
+				ones = in.readLine();
+			}
+			else{
+				nextLine = false;
+			}
+		}
```

Comment:
It doesn't seem to scan the file properly and it's so hard to understand what your code is doing.
Try to read yourself your code and check if it's written intuitively and readable by someone who doesn't understand code at all.